docs/proposals/gpu-support.md

```diff
@@ -0,0 +1,305 @@
+control ones like `nvidiactl`
+
+The `nvidia-docker` wrapper also verifies that the CUDA version required by a
+given image is supported by the host drivers, through inspection of well-known
+image labels, if present. We should try to provide equivalent checks.
+
+This is current sample output from the plugin, wrapped for readability:
+
+    $ curl -s localhost:3476/docker/cli
+    --device=/dev/nvidiactl --device=/dev/nvidia-uvm --device=/dev/nvidia0
+    --device=/dev/nvidia1 --volume-driver=nvidia-docker
+    --volume=nvidia_driver_352.68:/usr/local/nvidia:ro
+
+An obvious change is an additional endpoint for JSON output. The unresolved
+question is whether the plugin should be run standalone like that (called over
+HTTP, perhaps with endpoints for a new Kubernetes resource API) or whether the
+relevant code from its `nvidia` package should be linked directly into kubelet.
+A partial list of tradeoffs:
+
+|                     | External binary                                                                                   | Linked in                                                    |
+|---------------------|---------------------------------------------------------------------------------------------------|--------------------------------------------------------------|
+| Use of cgo          | Confined to binary                                                                                | Linked into kubelet, but with lazy binding                   |
+| Expandibility       | Limited if we run the plugin, increased if library is used to build a Kubernetes-tailored daemon. | Can reuse the `nvidia` library as we prefer                  |
+| Bloat               | None                                                                                              | Larger kubelet, even for systems without GPUs                |
+| Reliability         | Need to handle the binary disappearing at any time                                                | Fewer headeaches                                             |
+| (Un)Marshalling     | Need to talk over JSON                                                                            | None                                                         |
+| Administration cost | One more daemon to install, configure and monitor                                                 | No extra work required, other than perhaps configuring flags |
+| Releases            | Potentially on its own schedule                                                                   | Tied to Kubernetes'                                          |
+
+For all its drawbacks, the external binary solution is probably a better fit for
+V0.
+
+However we obtain the data, we'll need to change Kubelet's `dockertools` to also
+map devices into the containers and manage the CUDA volumes. As shown above,
+volumes are named after the hardware driver version.
+
+## Timeline
```

Comment:
I thought I would revisit the document later, which is why only V0 is detailed. I removed release targets and changed the title to Implementation Plan.